templates/pkg/resource/sdk.go.tpl

```diff
@@ -164,11 +164,16 @@ func (rm *resourceManager) updateConditions (
 
 	// Terminal condition
 	var terminalCondition *ackv1alpha1.Condition = nil
+	var recoverableCondition *ackv1alpha1.Condition = nil
 	for _, condition := range ko.Status.Conditions {
 		if condition.Type == ackv1alpha1.ConditionTypeTerminal {
 			terminalCondition = condition
 			break
```

Comment:
break should be removed

Reply:
Thanks for catching this. break causes the RecoverableError to be added twice. Removing it.